sky/task.py

```diff
@@ -484,13 +489,10 @@ def from_yaml_config(
                 'experimental.config_overrides')
             resources_config[
                 '_cluster_config_overrides'] = cluster_config_override
+        if task.vpn_config is not None:
+            resources_config.pop('ports', None)
```

Comment:
I dont think we should pop the resources section as a whole - this will also be displayed at `sky status` and user might get confused about it. Could we change our provision logic instead?

Reply:
Bump on this comment: it seems like we are still not showing the ports field in the resources after provisioning.

```bash
(sky-serve) ➜  skypilot git:(vpn-enhanced) sky launch @temp/vpn.yaml
Task from YAML spec: @temp/vpn.yaml
Considered resources (1 node):
----------------------------------------------------------------------------------------
 CLOUD   INSTANCE    vCPUs   Mem(GB)   ACCELERATORS   REGION/ZONE   COST ($)   CHOSEN   
----------------------------------------------------------------------------------------
 AWS     m6i.large   2       8         -              us-east-1     0.10          ✔     
----------------------------------------------------------------------------------------
Launching a new cluster 'sky-8672-txia'. Proceed? [Y/n]: 
⚙︎ Launching on AWS us-east-1 (us-east-1a,us-east-1b,us-east-1c,us-east-1d,us-east-1f).
└── Instance is up.
⠧ Preparing SkyPilot runtime (3/3 - runtime)  View logs at: ~/sky_logs/sky-2024-12-21-03-18
⠏ Preparing SkyPilot runtime (3/3 - runtime)  View logs at: ~/sky_logs/sky-2024-12-21-03-18
⠋ Preparing SkyPilot runtime (3/3 - runtime)  View logs at: ~/sky_logs/sky-2024-12-21-03-18
✓ Cluster launched: sky-8672-txia.  View logs at: ~/sky_logs/sky-2024-12-21-03-18-33-905960/provision.log
Run commands not specified or empty.

Cluster name: sky-8672-txia
├── To log into the head VM:    ssh sky-8672-txia
├── To submit a job:            sky exec sky-8672-txia yaml_file
├── To stop the cluster:        sky stop sky-8672-txia
└── To teardown the cluster:    sky down sky-8672-txia

(sky-serve) ➜  skypilot git:(vpn-enhanced) sky status sky-8672-txia
Clusters
NAME           LAUNCHED        RESOURCES          STATUS  AUTOSTOP  COMMAND                    
sky-8672-txia  a few secs ago  1x AWS(m6i.large)  UP      -         sky launch @temp/vpn.yaml
```

```yaml
# @temp/vpn.yaml
resources:
  cloud: aws
  cpus: 2
  ports:
    - 8080
    - 7000-7034
    - 60000
vpn:
  tailscale: true
```